src/cdk/drag-drop/directives/drop-list.ts

```diff
@@ -0,0 +1,309 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {coerceArray, coerceBooleanProperty} from '@angular/cdk/coercion';
+import {
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Input,
+  OnDestroy,
+  Output,
+  QueryList,
+  Optional,
+  Directive,
+  ChangeDetectorRef,
+  SkipSelf,
+} from '@angular/core';
+import {Directionality} from '@angular/cdk/bidi';
+import {CdkDrag} from './drag';
+import {DragDropRegistry} from '../drag-drop-registry';
+import {CdkDragDrop, CdkDragEnter, CdkDragExit, CdkDragSortEvent} from '../drag-events';
+import {CDK_DROP_LIST_CONTAINER, CdkDropListContainer} from '../drop-list-container';
+import {CdkDropListGroup} from './drop-list-group';
+import {DropListRef} from '../drop-list-ref';
+import {DragRef} from '../drag-ref';
+
+/** Counter used to generate unique ids for drop zones. */
+let _uniqueIdCounter = 0;
+
+// tslint:disable:class-name
+// TODO: exclude from public-api.ts
```

Comment:
This is something we'd have to do before the commit lands.

Since it won't be exported at all, we could use a name like `CdkDropListInternal`. It would also be good to have some tooling that could _enforce_ that certain symbols are not exported through the public api.

Reply:
Isn't this the purpose of the `@internal` tag? I know that we removed them to avoid downstream issues, but the symbols should be preserved if we didn't enable `stripInternal`.